bazel/envoy_internal.bzl

```diff
@@ -52,6 +52,10 @@ def envoy_copts(repository, test = False):
            }) + select({
                repository + "//bazel:enable_log_debug_assert_in_release": ["-DENVOY_LOG_DEBUG_ASSERT_IN_RELEASE"],
                "//conditions:default": [],
+           }) + select({
+               # APPLE_USE_RFC_3542 is needed to support IPV6_PKTINFO in MAC OS.
```

Comment:
Nice, thanks for figuring this out. If it were me I would have just compiled all of this out on Apple. :)